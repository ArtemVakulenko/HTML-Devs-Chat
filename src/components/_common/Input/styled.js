import styled from 'styled-components';

export const StInput = styled.input`
  padding: 15px 10px;
  background: ${({ colors, theme }) => colors[theme].input};
  border-radius: 10px;
  outline: none;
  border: none;
  color: ${({ colors, theme }) => colors[theme].color};
  font-size: 14px;
  min-width: 290px;
  &::placeholder {
    color: ${({ colors, theme }) => colors[theme].placeholder};
}
`;

export const StHidePass = styled.img`
  position: absolute;
  right:10px;
  bottom: 12px;
  cursor: pointer;
`;
