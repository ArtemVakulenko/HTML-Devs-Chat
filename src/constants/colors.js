export const colors = {
  dark: {
    main: '#0E1621',
    secondary: '#17212B',
    input: '#242F3D',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#fff',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#0E1621',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#488ccf',
  },
  light: {
    main: '#FFFFFF',
    secondary: '#ECECEC',
    input: '#D1D1D1',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#000',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#FFFFFF',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#2B5278',
  },
};
