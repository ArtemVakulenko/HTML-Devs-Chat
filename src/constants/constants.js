export const langs = {
  arabian: 'ar',
  russian: 'ru',
  english: 'en',
};

export const cookieAge = 20000;
