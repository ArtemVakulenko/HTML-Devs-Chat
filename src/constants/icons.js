export const icons = {
    changeThemeIcon: '../assets/img/changeThemeIcon.png',
    defaultUserIcon: '../assets/img/defaultUserIcon.png',
    dropDownIcon: '../assets/img/dropDownIcon.png',
    logoIcon: '../assets/img/logoIcon.png',
    logOutIcon: '../assets/img/logOutIcon.png',
    settingIcon: '../assets/img/settingsIcon.png',
    closeModalDark: '../assets/img/closeModalDark.png',
    sendMessage: '../assets/img/sendMessage.png',
    showPass: '../assets/img/view.png',
    hidePass: '../assets/img/hide.png',
    themeDark: '../assets/img/moon.png',
    themeLight: '../assets/img/sun.png',
};
