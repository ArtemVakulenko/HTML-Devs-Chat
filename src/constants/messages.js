export const MESSAGES_KEYS = {
  invalidLogin: 'invalidLogin',
  invalidPassword: 'invalidPassword',
  invalidConfirm: 'invalidConfirm',
  userIsReg: 'userIsReg',
  notExist: 'notExist',
  wrongPass: 'wrongPass',
  abstract: 'something went wrong',
  successReg: 'successReg',
  successAuth: 'successAuth',
  unauth: 'unAuth',
};
