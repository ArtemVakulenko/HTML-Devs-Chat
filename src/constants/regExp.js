export const LOGIN = new RegExp(/^[a-z0-9]{3,18}/, 'i');
export const PASSWORD = new RegExp(/^[a-z0-9!@$^."â„–;%:?*\(\)-_=+]{5,20}$/, 'i');
