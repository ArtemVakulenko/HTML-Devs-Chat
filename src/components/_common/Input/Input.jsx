import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { icons } from 'src/constants/icons';
import { StInput, StHidePass } from './styled';
import { useTheme } from '../../hooks';

const Input = ({ value, placeholder, onChange, type = 'text' }) => {
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  const [passSettig, setPassSettig] = useState({
    type: 'password',
    eye: icons.showPass,
  });
  const handleInputChange = (e) => {
    onChange(e.target.value);
  };
  const showHidePass = () => {
    passSettig.type === 'password'
      ? setPassSettig({ type: 'text', eye: icons.hidePass })
      : setPassSettig({ type: 'password', eye: icons.showPass });
  };
  return (
    <>
      <StInput
        type={type !== 'password' ? type : passSettig.type}
        value={value}
        theme={theme}
        colors={colors}
        onChange={handleInputChange}
        placeholder={t(placeholder)}
      />
      {type === 'password' ? <StHidePass src={passSettig.eye} onClick={showHidePass} /> : null}
    </>
  );
};

export default Input;

Input.propTypes = {
  type: PropTypes.string,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};
