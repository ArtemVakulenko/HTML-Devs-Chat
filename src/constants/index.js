import { colors } from './colors';
import { icons } from './icons';
import { MESSAGES_KEYS } from './messages';
import { url } from './url';
import { LOGIN, PASSWORD } from './regExp';
import { langs, cookieAge } from './constants';

export { url, colors, icons, MESSAGES_KEYS, LOGIN, PASSWORD, langs, cookieAge };
