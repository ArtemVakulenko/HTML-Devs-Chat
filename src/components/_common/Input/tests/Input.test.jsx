import { shallow, mount } from 'enzyme';
import React from 'react';
import Input from '../Input';
import { icons } from '../../../../constants';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
describe('Input', () => {
  describe('type text', () => {
    const props = {
      value: 'MockValue',
      placeholder: 'MockPlacholder',
      onChange: jest.fn(),
      type: 'text',
    };
    it('Should match snapshot', () => {
      const component = shallow(<Input {...props} />);
      expect(component.html()).toMatchSnapshot();
    });
    it('Should have StInput', () => {
      const component = mount(<Input {...props} />);
      expect(component.find('styled__StInput')).toHaveLength(1);
    });
    it('Should be a change', () => {
      const newValue = 'MockValueNew';
      const component = mount(<Input {...props} />);
      component.find('input').simulate('change', { target: { value: newValue } });
      expect(component.props().onChange).toHaveBeenCalledWith(newValue);
    });
  });
  describe('type password', () => {
    const props = {
      value: 'MockValue',
      placeholder: 'MockPlacholder',
      onChange: jest.fn(),
      type: 'password',
    };
    it('Should match snapshot', () => {
      const component = shallow(<Input {...props} />);
      expect(component.html()).toMatchSnapshot();
    });
    it('Should have StInput', () => {
      const component = mount(<Input {...props} />);
      expect(component.find('styled__StInput')).toHaveLength(1);
    });
    it('Should have StHidePass', () => {
      const component = mount(<Input {...props} />);
      expect(component.find('styled__StHidePass')).toHaveLength(1);
    });
    it('Should be a change', () => {
      const newValue = 'MockValueNew';
      const component = mount(<Input {...props} />);
      component.find('input').simulate('change', { target: { value: newValue } });
      expect(component.props().onChange).toHaveBeenCalledWith(newValue);
    });
    it('Should be a eye click', () => {
      const component = mount(<Input {...props} />);
      const eye = component.find('styled__StHidePass').simulate('click');
      expect(eye.props().src).toBe(icons.showPass);
    });
    it('Should be a eye two click', () => {
      const component = mount(<Input {...props} />);
      const eye = component.find('styled__StHidePass').simulate('click');
      eye.simulate('click');
      expect(eye.props().src).toBe(icons.showPass);
    });
  });
});
