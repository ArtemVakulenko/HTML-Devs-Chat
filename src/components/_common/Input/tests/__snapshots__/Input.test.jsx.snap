// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input type password Should match snapshot 1`] = `"<input type=\\"password\\" value=\\"MockValue\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/><img src=\\"../assets/img/view.png\\" class=\\"styled__StHidePass-kZfxvt kfuTnG\\"/>"`;

exports[`Input type text Should match snapshot 1`] = `"<input type=\\"text\\" value=\\"MockValue\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/>"`;
